table/update_schema.go

```diff
@@ -0,0 +1,834 @@
+		var err error
+		out, err = reorder(out, reqs)
+		if err != nil {
+			return nil, fmt.Errorf("error reordering fields: %w", err)
+		}
+	}
+
+	return out, nil
+}
+
+// Reorder fields based on move operations
+func reorder(fields []iceberg.NestedField, reqs []moveReq) ([]iceberg.NestedField, error) {
+	// find the index of a field by its id
+	indexOf := func(id int) int {
+		for i, f := range fields {
+			if f.ID == id {
+				return i
+			}
+		}
+		return -1
+	}
+
+	for _, m := range reqs {
+		pos := indexOf(m.fieldID)
+		if pos == -1 {
+			continue // field might have been deleted
+		}
+
+		f := fields[pos]
+		fields = append(fields[:pos], fields[pos+1:]...)
+
+		switch m.op {
+		case OpFirst:
+			fields = append([]iceberg.NestedField{f}, fields...)
+		case OpBefore:
+			idx := indexOf(m.otherFieldID)
+			if idx == -1 {
+				return nil, errors.New("move-before target not found at commit time")
+			}
+			fields = append(fields[:idx],
+				append([]iceberg.NestedField{f}, fields[idx:]...)...)
+		case OpAfter:
+			idx := indexOf(m.otherFieldID)
+			if idx == -1 {
+				return nil, errors.New("move-after target not found at commit time")
+			}
+			fields = append(fields[:idx+1],
+				append([]iceberg.NestedField{f}, fields[idx+1:]...)...)
+		default:
+			return nil, fmt.Errorf("unknown move op: %s", m.op)
+		}
+	}
+
+	return fields, nil
+}
+
+// Helper functions
+func findField(schema *iceberg.Schema, path []string, caseSensitive bool) *iceberg.NestedField {
+	name := strings.Join(path, ".")
+
+	var field iceberg.NestedField
+	var ok bool
+
+	if caseSensitive {
+		field, ok = schema.FindFieldByName(name)
+	} else {
+		field, ok = schema.FindFieldByNameCaseInsensitive(name)
+	}
+
+	if !ok {
+		return nil
+	}
+
+	return &field
+}
+
+func parentIDForPath(schema *iceberg.Schema, path []string, caseSensitive bool) int {
+	if len(path) == 1 {
+		return -1
+	}
+
+	if f := findField(schema, path[:len(path)-1], caseSensitive); f != nil {
+		return f.ID
+	}
+
+	return -1
+}
+
+func validateDefaultValue(typ iceberg.Type, val any) error {
+	// Defaults are only allowed on primitive columns
+	prim, ok := typ.(iceberg.PrimitiveType)
+	if !ok {
+		return fmt.Errorf("defaults are only allowed on primitive columns, got %s", typ.Type())
+	}
+
+	lit, err := iceberg.LiteralFromAny(val)
+	if err != nil {
+		return err
+	}
+
+	litType := lit.Type()
+
+	// Exact match ?
+	if litType.Equals(prim) {
+		return nil
+	}
+
+	return fmt.Errorf("default literal of type %s is not assignable to of type %s", litType.String(), prim.Type())
+}
+
+func (us *UpdateSchema) findExistingSchemaInTransaction(newSchema *iceberg.Schema) *int {
+	for _, schema := range us.txn.tbl.metadata.Schemas() {
+		if newSchema.Equals(schema) {
+			return &schema.ID
+		}
+	}
+	return nil
+}
+
+func (us *UpdateSchema) getNameMappingUpdates(newSchema *iceberg.Schema) []Update {
+	// Only update name mapping if we have adds/updates that might need it
+	hasAdds := false
+	hasUpdates := false
```

Comment:
we're not doing different behaviors based on `hasAdds` vs `hasUpdates` so just use one variable